src/juju/actions.js

```diff
@@ -0,0 +1,29 @@
+// Action labels
+export const actionsList = {
+  fetchModelList: "FETCH_MODEL_LIST",
+  updateModelList: "UPDATE_MODEL_LIST"
+};
+
+// Action creators
+/**
+  @param {Array} models The list of models to store.
+*/
+export function updateModelList(models) {
+  return {
+    type: actionsList.updateModelList,
+    payload: models
+  };
+}
+
+/**
+  Fetches the model list from the supplied Juju controller.
+  @returns {Object} models The list of model objects under the key `userModels`.
+*/
+export function fetchModelList() {
+  return async function(dispatch, getState) {
+    const conn = getState().root.controllerConnection;
```

Comment:
You might find these posts interesting:

* https://blog.isquaredsoftware.com/2017/01/idiomatic-redux-thoughts-on-thunks-sagas-abstraction-and-reusability/
* https://stackoverflow.com/questions/35667249/accessing-redux-state-in-an-action-creator/35674575#35674575

Generally using getState in thunks is considered an anti-pattern apparently (and certainly is in an action creator), but I think this meets the criteria Dan mentions in the SO post, injecting some dependent cached data, in this case a connection.

Reply:
Yep I agree, thanks for the links, some good content there!